### PR TITLE
Add rust-postgres to clients test suite.

### PR DESCRIPTION
**Pull request by kelvich**
Also pass NEON_CONNSTR to that tests. IMO it is a bit more illustrative to have examples using connection string when possible.

### CODE REVIEW

**Review by bayandin — APPROVED** [review]
Looks good; just a couple of comments:
- In pg_clients directory we have `language/library` structure, so the example should be moved to `rust/postgres` directory
- You need explicitly add `rust/postgres` to `@pytest.mark.parametrize` to run it nightly

File: test_runner/pg_clients/rust/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM rust:1.66 AS build
+WORKDIR /source
+COPY . .
+RUN cargo build --release
+
+FROM rust:1.66
```
Comment (vadim2404) [inline test_runner/pg_clients/rust/Dockerfile:6]: you can derive from some minimalistic image without rust at all. For instance, compute uses `FROM debian:bullseye-slim`

File: test_runner/pg_clients/rust/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+use std::env;
+
+fn main() -> Result<(), Box<dyn error::Error>> {
+    let builder = SslConnector::builder(SslMethod::tls())?;
+    let connector = MakeTlsConnector::new(builder.build());
+
+    let mut client = Client::connect(&env::var("NEON_CONNSTR")?, connector)?;
+
+    for row in client.query("SELECT 42", &[])? {
+        let ret : i32 = row.get(0);
```
Comment (vadim2404) [inline test_runner/pg_clients/rust/src/main.rs:14]: NIT:

```rust
let ret: i32
```

removed space before `:`

**Review by vadim2404 — APPROVED** [review]

File: test_runner/pg_clients/rust/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+[package]
+name = "rust-neon"
+version = "0.1.0"
+edition = "2021"
+
```
Comment (vadim2404) [inline test_runner/pg_clients/rust/Cargo.toml:5]: add `publish = false` as a precaution

**Comment by hlinnaka** [comment]
Superseded by https://github.com/neondatabase/neon/pull/3750, closing.

> Also pass NEON_CONNSTR to that tests. IMO it is a bit more illustrative to have examples using connection string when possible.

I didn't do that in PR #3750 (I didn't notice this PR until now). I think that's a good idea, though.

**Pull request closed without merging**